src/hist.jl

```diff
@@ -14,6 +16,31 @@ function _check_closed_arg(closed::Symbol, funcsym)
 end
 
 
+## Fast map for tuples (a lot faster than using Base.map)
+@generated function _tuple_map{N}(f, tuples::NTuple{N, Any}...)
+    M = length(typeof(tuples::NTuple).parameters)
+    result_expr = Expr(:tuple)
+    for i in 1:N
+        call_f_expr = Expr(:call,:f)
+        for j in 1:M
+            push!(call_f_expr.args, :(tuples[$j][$i]))
+        end
+        push!(result_expr.args, call_f_expr)
+    end
+    result_expr
+end
+
+## Fast getindex function for multiple arrays, returns a tuple of array elements
+@inline Base.@propagate_inbounds @generated function _multi_getindex(i::Integer, c::AbstractArray...)
+    N = length(typeof(c::Tuple).parameters)
```

Comment:
Uh, yeah - this went through several iterations ... anyhow, I replaced it with `length(c)` as suggested. Question - should I move this into "common.jl"? This one is definitely very relevant for performance - maybe it could be used in other places in StatsBase at some point?